### PR TITLE
Print exception to a console when debugger paused

### PR DESCRIPTION
**Pull request by TatriX**
This is a fast hacked version of what I want to see when debugger is paused on exception.
Currently I can see a line where error occurred but the only way to know what's wrong is to press "c" and then get an error message in the console.

### CODE REVIEW

File: indium-webkit.el [hunk 1]
```diff
@@ -347,8 +347,14 @@ MESSAGE explains why the connection has been closed."
...
-        (exception (equal (map-nested-elt message '(params reason)) "exception"))
-        (reason (if exception "Exception occured" "Breakpoint hit")))
+         (exception (equal (map-nested-elt message '(params reason)) "exception"))
+         (params (map-nested-elt message '(params)))
+         (reason (if exception "Exception occured" "Breakpoint hit")))
+
+    (when exception
+      (indium-repl-emit-console-message (setf (map-elt params 'values)
+                                              (list (indium-webkit--value (map-elt params 'data))))
+                                        t))
```
Comment (NicolasPetton) [inline indium-webkit.el]: Could you do the same thing in `indium-v8-inspector.el`?

File: indium-v8-inspector.el [hunk 2]
```diff
@@ -290,8 +290,13 @@ the connection and buffers."
 
 (defun indium-v8-inspector--handle-debugger-paused (message)
   (let* ((frames (map-nested-elt message '(params callFrames)))
-        (exception (equal (map-nested-elt message '(params reason)) "exception"))
-        (reason (if exception "Exception occured" "Breakpoint hit")))
+         (exception (equal (map-nested-elt message '(params reason)) "exception"))
+         (params (map-nested-elt message '(params)))
+         (reason (if exception "Exception occured" "Breakpoint hit")))
+    (when exception
+      (indium-repl-emit-console-message (setf (map-elt params 'values)
```
Comment (NicolasPetton) [inline indium-v8-inspector.el]: Why do you mutate `params`?

Comment (TatriX) [inline indium-v8-inspector.el]: Copied from `indium-webkit--exception`.

Comment (TatriX) [inline indium-v8-inspector.el]: Should I change to a pure code?

**Comment by NicolasPetton** [comment]
TatriX <notifications@github.com> writes:

> Should I change to a pure code?

Unless you need to mutate `params` for a reason, I think that'd be cleaner.

**Comment by NicolasPetton** [comment]
@TatriX I gave your branch a try, but I don't like having the exception printed in the REPL before it's actually thrown. Also, the user will get it twice now if he proceeds the execution.

I think the proper way to solve the issue (being able to know more about the exception when entering the debugger) would be to improve the debugger's UI to be able to show a description of the exception.

I'll make some experiments and see if I can come up with something good.

**Comment by TatriX** [comment]
At least now I can see what's wrong. It wasn't possible before. So I can live with duplicated messages.

**Comment by TatriX** [comment]
Probably "Debugger stack" is nice place to add error description?

**Comment by NicolasPetton** [comment]
What about something like this:

screenshot from 2017-05-15 22-04-41

/cc @DamienCassou what do you think? This were an issue for you as well, right?

**Comment by TatriX** [comment]
Looks good, but if possible I'd prefer not wasting another line.

**Comment by DamienCassou** [comment]
I miss this message indeed. Both solutions are good for me.

**Comment by NicolasPetton** [comment]
Thank you for working on this.  However, I think something a little fancier like displaying the error message inline would be awesome and give better context information since it's right where the exception is thrown.

screenshot from 2017-05-19 17-26-26
screenshot from 2017-05-19 17-26-17

If it doesn't work as well as I hope it will, I'll reopen this PR!

**Pull request closed without merging**